frontend/app/tracking/ActivityTracking.scala

```diff
@@ -0,0 +1,118 @@
+  def toMap: JMap[String, Object] =
+    Map("eventSource" -> source) ++ member.toMap
+}
+
+
+trait TrackerData {
+  def member: MemberData
+  def source: String
+  def toMap: JMap[String, Object]
+}
+
+sealed trait TierAmendment {
+  val tierFrom: Tier
+  val tierTo: Tier
+  val effectiveFromDate: Option[DateTime]
+}
+
+case class UpgradeAmendment(tierFrom: Tier, tierTo: Tier, effectiveFromDate: Option[DateTime] = Some(DateTime.now)) extends TierAmendment
+case class DowngradeAmendment(tierFrom: Tier, tierTo: Tier = Tier.Friend, effectiveFromDate: Option[DateTime] = None) extends TierAmendment
+
+case class MemberData(salesforceContactId: String,
+                        identityId: String,
+                        tier: String,
+                        tierAmendment: Option[TierAmendment] = None,
+                        deliveryPostcode: Option[String] = None,
+                        billingPostcode: Option[String] = None,
+                        subscriptionPaymentAnnual: Option[Boolean] = None,
+                        marketingChoices: Option[MarketingChoicesForm] = None) {
+
+  val subscriptionPlan = subscriptionPaymentAnnual match {
+    case Some(true) =>  Some("annual")
+    case Some(false) => Some("monthly")
+    case None => None
+  }
+
+  def toMap: JMap[String, Object] = {
+
+    def bcrypt(string: String) = (string+Config.bcryptPepper).bcrypt(Config.bcryptSalt)
+
+    val dataMap =
+      Map(
+        "salesforceContactId" -> bcrypt(salesforceContactId),
+        "identityId" -> bcrypt(identityId),
+        "tier" -> tier
+      ) ++
+        deliveryPostcode.map("deliveryPostcode" -> truncatePostcode(_)) ++
+        billingPostcode.map("billingPostcode" -> truncatePostcode(_)) ++
+        subscriptionPlan.map("subscriptionPlan" -> _) ++
+        marketingChoices.map { mc =>
+          "marketingChoicesForm" -> ActivityTracking.setSubMap {
+            Map(
+              "gnm" -> mc.gnm.getOrElse(false),
+              "thirdParty" -> mc.thirdParty.getOrElse(false),
+              "membership" -> true
+            )
+          }
+        } ++
+        tierAmendment.map { tierAmend =>
+          "amendTier" -> ActivityTracking.setSubMap {
+            Map(
+              "from" -> tierAmend.tierFrom.name,
+              "to" -> tierAmend.tierTo.name
+            ) ++
+            tierAmend.effectiveFromDate.map("effectiveDate" -> _)
+          }
+        }
+
+    ActivityTracking.setSubMap(dataMap)
+  }
+
+  def truncatePostcode(postcode: String) = {
+    postcode.splitAt(postcode.length-3)._1.trim
+  }
+}
+
+trait ActivityTracking {
+
+  def track(data: TrackerData) {
+    try {
+      val tracker = getTracker
+      val dataMap = data.toMap
+      tracker.trackUnstructuredEvent(dataMap)
+    } catch {
+      case error: Throwable =>
+      Logger.error(s"Activity tracking error: ${error.getMessage}")
+    }
+  }
+
+  private def getTracker: Tracker = {
+    val emitter = new Emitter(ActivityTracking.url, HttpMethod.GET)
```

Comment:
We can enable the async client with `emitter.setRequestMethod(RequestMethod.Asynchronous)`